tangelo/problem_decomposition/tests/dmet/test_osdmet.py

```diff
@@ -67,6 +67,23 @@ def test_lio2_sto6g_uhf(self):
 
         self.assertAlmostEqual(energy, -156.6243118102, places=4)
 
+    def test_notimplemented_uhf_hf_solver(self):
+        """Tests if setting UHF=True raises an error if a HF solver is selected."""
```

Comment:
Just to make sure I understand:

- We do not have a `HFSolver` class. Do you think we should have one in the future ? (how much work is it?)
- Does the `NotImplemented` error have anything to do with DMET in the end, or is it just a HF thing ?

Reply:
- Maybe, for now the `IntegralSolver` classes are kind of occupying this spot. This would require a bit of work, and mainly on the software design side of things. It is not trivial to move things and link them again (i.e. `SecondQuantizedMolecule` + `IntegralSolver.compute_mean_field()`).
- It is a mixed of DMET coding that does not support this as of now + untested science that I am not sure if this is valid. I don't have a use-case as of now to confirm if a solution, when the code will work, is correct.

Reply:
Thank you. Since "this is complicated", we'll just leave it as that.